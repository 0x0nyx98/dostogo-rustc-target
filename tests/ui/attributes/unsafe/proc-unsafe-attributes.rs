#![feature(unsafe_attributes)]

#[unsafe(proc_macro)]
//~^ ERROR: is not an unsafe attribute
//~| ERROR attribute is only usable with crates of the `proc-macro` crate type
pub fn a() {}


#[unsafe(proc_macro_derive(Foo))]
//~^ ERROR: is not an unsafe attribute
//~| ERROR attribute is only usable with crates of the `proc-macro` crate type
pub fn b() {}

#[proc_macro_derive(unsafe(Foo))]
//~^ ERROR attribute is only usable with crates of the `proc-macro` crate type
pub fn c() {}

#[unsafe(proc_macro_attribute)]
//~^ ERROR: is not an unsafe attribute
//~| ERROR attribute is only usable with crates of the `proc-macro` crate type
pub fn d() {}

#[unsafe(allow(dead_code))]
//~^ ERROR: is not an unsafe attribute
pub fn e() {}

fn main() {}
