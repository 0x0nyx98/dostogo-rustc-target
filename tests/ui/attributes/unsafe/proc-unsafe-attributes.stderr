error: `proc_macro` is not an unsafe attribute
  --> $DIR/proc-unsafe-attributes.rs:3:3
   |
LL | #[unsafe(proc_macro)]
   |   ^^^^^^
   |
   = note: extraneous unsafe is not allowed in attributes

error: `proc_macro_derive` is not an unsafe attribute
  --> $DIR/proc-unsafe-attributes.rs:9:3
   |
LL | #[unsafe(proc_macro_derive(Foo))]
   |   ^^^^^^
   |
   = note: extraneous unsafe is not allowed in attributes

error: `proc_macro_attribute` is not an unsafe attribute
  --> $DIR/proc-unsafe-attributes.rs:18:3
   |
LL | #[unsafe(proc_macro_attribute)]
   |   ^^^^^^
   |
   = note: extraneous unsafe is not allowed in attributes

error: `allow` is not an unsafe attribute
  --> $DIR/proc-unsafe-attributes.rs:23:3
   |
LL | #[unsafe(allow(dead_code))]
   |   ^^^^^^
   |
   = note: extraneous unsafe is not allowed in attributes

error: the `#[proc_macro]` attribute is only usable with crates of the `proc-macro` crate type
  --> $DIR/proc-unsafe-attributes.rs:3:1
   |
LL | #[unsafe(proc_macro)]
   | ^^^^^^^^^^^^^^^^^^^^^

error: the `#[proc_macro_derive]` attribute is only usable with crates of the `proc-macro` crate type
  --> $DIR/proc-unsafe-attributes.rs:9:1
   |
LL | #[unsafe(proc_macro_derive(Foo))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: the `#[proc_macro_derive]` attribute is only usable with crates of the `proc-macro` crate type
  --> $DIR/proc-unsafe-attributes.rs:14:1
   |
LL | #[proc_macro_derive(unsafe(Foo))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: the `#[proc_macro_attribute]` attribute is only usable with crates of the `proc-macro` crate type
  --> $DIR/proc-unsafe-attributes.rs:18:1
   |
LL | #[unsafe(proc_macro_attribute)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 8 previous errors

